forc-plugins/forc-crypto/src/keys/vanity.rs

```diff
@@ -0,0 +1,440 @@
+use anyhow::{anyhow, Result};
+use fuel_crypto::{fuel_types::Address, PublicKey, SecretKey};
+use fuels_accounts::wallet::{generate_mnemonic_phrase, DEFAULT_DERIVATION_PATH_PREFIX};
+use fuels_core::types::bech32::{Bech32Address, FUEL_BECH32_HRP};
+use rayon::iter::{self, Either, ParallelIterator};
+use regex::Regex;
+use serde_json::json;
+use std::{fs, path::PathBuf, time::Instant};
+
+forc_util::cli_examples! {
+    crate::Command {
+        [ Generate a checksummed vanity address with a given prefix => "forc crypto vanity --starts-with aaa" ]
+    }
+}
+
+#[derive(Debug, clap::Parser)]
+#[clap(
+    version,
+    about = "Generate a vanity address",
+    after_help = "Generate vanity addresses for the Fuel blockchain"
+)]
+pub struct Arg {
+    /// Prefix regex pattern or hex string.
+    #[arg(long, value_name = "PATTERN", required_unless_present = "ends_with")]
+    pub starts_with: Option<String>,
+
+    /// Suffix regex pattern or hex string.
+    #[arg(long, value_name = "PATTERN")]
+    pub ends_with: Option<String>,
+
+    /// Return mnemonic with address (default false)
+    #[arg(long)]
+    pub mnemonic: bool,
+
+    /// Path to save the generated vanity address to.
+    #[arg(long, value_hint = clap::ValueHint::FilePath, value_name = "PATH")]
+    pub save_path: Option<PathBuf>,
+}
```

Comment:
perhaps a timeout option could be nice? 

```rust
#[arg(long, value_name = "SECONDS")]
pub timeout: Option<u64>,
```

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6661/commits/723db3ee288d309baab22f151b656246c955bb27